haskell-src/Concordium/Types.hs

```diff
@@ -996,6 +1012,7 @@ instance S.Serialize PayloadSize where
 -- | Serialized payload of the transaction
 newtype EncodedPayload = EncodedPayload {_spayload :: BSS.ShortByteString}
     deriving (Eq, Show)
+    deriving (AE.ToJSON, AE.FromJSON) via BSH.ByteStringHex
```

Comment:
What is this needed for?

Reply:
Initially, the encoded payload was part of a partially signed transaction in the JSON file. Now we have changed it to use the decoded (human-readable) payload instead. So you are right that this line is currently not needed. I could remove it but since it was already implemented, I just kept it in case we need it at some point.